### ISSUE TITLE
Bug or undocumented behavior: Creation of TensorBlock moves Labels device to cpu

### ISSUE THREAD

**Issue by agoscinski** [lab-cosmo/equistore#290]
Hope it is clear from title, here code snippet

```python
import torch
from equistore.torch import Labels, TensorBlock

sample_labels = Labels(["s"], torch.tensor([[0], [2], [4]], device='cuda:0'))
print(sample_labels.values.device) # cuda:0
block_1 = TensorBlock(
    values=torch.full((3, 1, 1), 1.0, device='cuda:0'),
    samples=sample_labels,
    components=[Labels(["c"], torch.tensor([[0]], device='cuda:0'))],
    properties=Labels(["p"], torch.tensor([[0]], device='cuda:0')),
)
print(block_1.samples.values.device) # cpu
```

**Comment by Luthaf** [comment]
So the data needs to move to CPU for equistore to understand it, but we should also keep the initial copy on GPU and make sure it is accessible. I'll fix this!
